ocean_lib/models/datatoken_enterprise.py

```diff
@@ -77,15 +85,18 @@ def buy_DT_and_order(
     @enforce_types
     def dispense_and_order(
         self,
-        consumer: str,
-        service_index: int,
         provider_fees: dict,
         tx_dict: dict,
+        consumer: Optional[str] = None,
+        service_index: int = 1,
```

Comment:
The index is not by default 0?

Reply:
I think there's 1-based indexing. I just transfered it here from the tests.

Reply:
And I think it's a remnant from when we didn't have string-based ids, so indices started at 1 because identifying with 0 is a bit weird